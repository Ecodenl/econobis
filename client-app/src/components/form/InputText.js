import React from 'react';
import PropTypes from 'prop-types';

const InputText = props => {
    const {
        label,
        type,
        className,
        size,
        id,
        placeholder,
        name,
        value,
        onClickAction,
        onChangeAction,
        onBlurAction,
        required,
        readOnly,
        maxLength,
        error,
        min,
        max,
        step,
        errorMessage,
        divSize,
        divClassName,
        autoComplete,
    } = props;

    return (
        <div className={`form-group ${divSize} ${divClassName}`}>
            <label htmlFor={id} className={`col-sm-6 ${required}`}>
                {label}
            </label>
            <div className={`${size}`}>
                <input
                    type={type}
                    className={`form-control input-sm ${className}` + (error ? 'has-error' : '')}
                    id={id}
                    placeholder={placeholder}
                    name={name}
                    value={value}
                    onClick={onClickAction}
                    onChange={onChangeAction}
                    onBlur={onBlurAction}
                    readOnly={readOnly}
                    maxLength={maxLength}
                    min={min}
                    max={max}
                    autoComplete={autoComplete}
                    step={step}
                />
            </div>
            {error && (
                <div className="col-sm-offset-6 col-sm-6">
                    <span className="has-error-message"> {errorMessage}</span>
                </div>
            )}
        </div>
    );
};

InputText.defaultProps = {
    divClassName: '',
    className: '',
    size: 'col-sm-6',
    divSize: 'col-sm-6',
    type: 'text',
    value: '',
    required: '',
    readOnly: false,
    maxLength: null,
    error: false,
    min: '',
    max: '',
    step: '',
    errorMessage: '',
    autoComplete: 'off',
    onBlurAction: () => {},
    onClickAction: () => {},
    onChangeAction: () => {},
};

InputText.propTypes = {
    label: PropTypes.oneOfType([PropTypes.string, PropTypes.object]).isRequired,
    type: PropTypes.string,
    className: PropTypes.string,
    divClassName: PropTypes.string,
    size: PropTypes.string,
    divSize: PropTypes.string,
    id: PropTypes.string,
    placeholder: PropTypes.string,
    name: PropTypes.string.isRequired,
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    onClickAction: PropTypes.func,
    onChangeAction: PropTypes.func,
    onBlurAction: PropTypes.func,
    required: PropTypes.string,
    readOnly: PropTypes.bool,
    maxLength: PropTypes.string,
    error: PropTypes.bool,
    min: PropTypes.string,
    max: PropTypes.string,
    step: PropTypes.string,
    errorMessage: PropTypes.string,
    autoComplete: PropTypes.string,
};

export default InputText;
