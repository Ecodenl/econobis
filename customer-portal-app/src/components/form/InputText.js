import React from 'react';
import PropTypes from 'prop-types';
import { get } from 'lodash';

const InputText = ({
    field,
    type,
    className,
    id,
    readOnly,
    placeholder,
    showErrorMessage,
    errors,
    touched,
    classNameErrorMessage,
    disabled,
}) => {
    return (
        <>
            {get(errors, field.name, '') && get(touched, field.name, '') && showErrorMessage ? (
                <small className={`${classNameErrorMessage}`}>{get(errors, field.name, '')}</small>
            ) : null}
            <input
                type={type}
                className={`text-input w-input ${className} ${
                    Boolean(get(errors, field.name, '') && get(touched, field.name, '')) ? 'has-error mb-0' : ''
                } `}
                id={id}
                {...field}
                readOnly={readOnly}
                placeholder={placeholder}
                disabled={disabled}
            />
        </>
    );
};

InputText.defaultProps = {
    type: 'text',
    className: 'content',
    readOnly: false,
    placeholder: '',
    showErrorMessage: true,
    classNameErrorMessage: 'text-danger',
    errors: {},
    touched: {},
};

InputText.propTypes = {
    field: PropTypes.object.isRequired,
    type: PropTypes.string,
    className: PropTypes.string,
    id: PropTypes.string,
    readOnly: PropTypes.bool,
    showErrorMessage: PropTypes.bool,
    classNameErrorMessage: PropTypes.string,
    placeholder: PropTypes.string,
    errors: PropTypes.object,
    touched: PropTypes.object,
};

export default InputText;
