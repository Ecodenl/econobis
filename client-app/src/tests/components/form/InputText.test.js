import React from 'react';
import { shallow } from 'enzyme';
import InputText from '../../../components/form/InputText';

test('should render InputText correctly', () => {
    const wrapper = shallow(<InputText label={"Naam"} name={"naam"} value={"Rob"}/>);
    expect(wrapper).toMatchSnapshot();
});