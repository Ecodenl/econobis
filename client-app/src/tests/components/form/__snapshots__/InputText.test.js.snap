// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render InputText correctly 1`] = `
<div
  className="form-group col-sm-6"
>
  <label
    className="col-sm-6 "
  >
    Naam
  </label>
  <div
    className="col-sm-6"
  >
    <input
      className="form-control input-sm false"
      maxLength={null}
      name="naam"
      readOnly={false}
      type="text"
      value="Rob"
    />
  </div>
</div>
`;
